pkg/multicluster/watcher.go

```diff
@@ -0,0 +1,111 @@
+package multicluster
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/go-utils/contextutils"
+	"github.com/solo-io/skv2/pkg/api/kube/core/v1/controller"
+	"github.com/solo-io/skv2/pkg/multicluster/kubeconfig"
+	"github.com/solo-io/skv2/pkg/reconcile"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	LocalCluster = ""
+)
+
+// AddClusterHandler is called when a new cluster is added.
+// The provided context is cancelled when a cluster is removed, so any teardown behavior for removed clusters
+// should take place when ctx is cancelled.
+type AddClusterHandler func(ctx context.Context, cluster string, mgr manager.Manager) error
+
+// ClusterWatcher calls ClusterHandlers and maintains a setManager of active cluster credentials.
+type ClusterWatcher interface {
+	controller.SecretReconciler
```

Comment:
what is the purpose of this interface? It's not clear to me why we would define an interface on top of SecretReconciler

Reply:
this is an artifact from a previous implementation, removing